rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -286,8 +286,8 @@ public synchronized void processBest(Block block) {
         logger.trace("Processing best block {} {}", block.getNumber(), block.getShortHash());
 
         if (bestBlock != null) {
-            BlockFork fork = new BlockFork();
-            fork.calculate(bestBlock, block, blockStore);
+            BlockchainBranchComparator branchComparator = new BlockchainBranchComparator(blockStore);
```

Comment:
It may be an over engineering, but I think it would make sense to inject this as a component

Reply:
The `BlockchainBranchComparator` doesn't hold state, and by injecting it we still don't get rid of the `blockStore` dependency.

Injecting it could make testing `TransactionPoolImpl` simpler, but since I won't do it now I'd leave it as is.